tests/kernel/mellanox_config.pm

```diff
@@ -19,64 +21,44 @@ use testapi;
 use utils;
 use ipmi_backend_utils;
 use power_action_utils 'power_action';
-use version_utils 'is_sle';
-
-our $device1 = '/dev/mst/mt4119_pciconf0';
-our $device2 = '/dev/mst/mt4119_pciconf0.1';
-
-sub show_links {
-    assert_script_run("mlxconfig -d $device1 q|grep LINK_TYPE");
-    assert_script_run("mlxconfig -d $device2 q|grep LINK_TYPE");
-}
 
 sub run {
     my $self = shift;
-    $self->select_serial_terminal;
+    my $protocol = get_var('MLX_PROTOCOL', 2);
 
-    my $mft_version = get_required_var('MFT_VERSION');
-    my $protocol    = get_var('MLX_PROTOCOL') || 2;
+    # allow to configure SR-IOV and enable virtual functions
+    my $sriov_en = get_var('MLX_SRIOV',   0);
+    my $num_vfs  = get_var('MLX_NUM_VFS', 0);
 
-    if (is_sle('>=15')) {
-        zypper_call("ar -f -G http://download.suse.de/ibs/SUSE:/SLE-15-SP1:/GA:/TEST/images/repo/SLE-15-SP1-Module-Development-Tools-POOL-x86_64-Media1/  dev_1");
-        zypper_call("ar -f -G http://download.suse.de/ibs/SUSE:/SLE-15-SP1:/GA:/TEST/images/repo/SLE-15-SP1-Module-Development-Tools-POOL-x86_64-Media2/  dev_2");
-        zypper_call("ar -f -G http://download.suse.de/ibs/SUSE:/SLE-15-SP1:/GA/standard/SUSE:SLE-15-SP1:GA.repo");
-    }
-    zypper_call('--quiet in kernel-source rpm-build wget pciutils', timeout => 200);
+    # right now, this code will only do something reasonable if we
+    # run on a baremetal machine (and thus on IPMI backend)
+    return unless check_var('BACKEND', 'ipmi');
 
-    # Install Mellanox Firmware Tool (MFT)
-    assert_script_run("wget http://www.mellanox.com/downloads/MFT/" . $mft_version . ".tgz");
-    assert_script_run("tar -xzvf " . $mft_version . ".tgz");
-    assert_script_run("rm " . $mft_version . ".tgz");
-    assert_script_run("./" . $mft_version . "/install.sh");
-    assert_script_run("mst start");
+    $self->select_serial_terminal;
+
+    # install dependencies
+    zypper_call('--quiet in pciutils mstflint', timeout => 200);
 
-    # List network devices
-    assert_script_run("lspci|egrep -i 'network|ethernet'");
+    my @devices = split(' ', script_output("lspci | grep -i Mellanox |cut  -d ' ' -f 1"));
 
-    if (check_var('BACKEND', 'ipmi')) {
-        if (script_run("lspci|grep -i mellanox") != 0) {
-            die "There is no Mellanox card here";
-        }
-        if (script_run("ls " . $device1) != 0) {
-            die "The directory $device1 doesn't exist";
-        }
-        if (script_run("ls " . $device2) != 0) {
-            die "The directory $device2 doesn't exist";
-        }
-        # Change Link protocol
-        record_info("INFO", "Wanted Link protocol is $protocol");
-        show_links();
-        assert_script_run("mlxconfig -y -d $device1 set LINK_TYPE_P1=$protocol LINK_TYPE_P2=$protocol");
-        assert_script_run("mlxconfig -y -d $device2 set LINK_TYPE_P1=$protocol LINK_TYPE_P2=$protocol");
-        show_links();
+    die "There is no Mellanox card here" if !@devices;
```

Comment:
in this case we could improve the output for debugging and do a record_info of the output of lspci command. I am ok if this is not done in this PR, but just as a suggestion to help debug.

Reply:
I can add the lspci output, but that will be rather lengthy and not contain useful information: if I grep for mellanox, it will be empty (that's what we get from the lspic | grep -i mellanox) or contain all other devices, which will be quite a lot on baremetal servers. I don't think either of those will be helpful, or am I missing something here?

Reply:
No, it's fine as it is. It's just a matter of adding more "potential" debug information in case it fails and the mellanox cards are inserted, but it's fine as well like this :)